src/mscorlib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -819,6 +819,11 @@ public unsafe virtual int IndexOf(string source, char value, int startIndex, int
             if (count < 0 || startIndex > source.Length - count)
                 throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_Count);
 
+            if (source.Length == 0)
```

Comment:
Is this the only overload that we need to add the sourc.Length check to?

Reply:
Yes, we have the check in place in all methods calling Last{IndexOfCore} which firing the assert.

Reply:
Should the check do something like this:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Globalization/CompareInfo.cs#L853-L860

Reply:
:-)

---
In reply to: [168056704](https://github.com/dotnet/coreclr/pull/16373#discussion_r168056704) [](ancestors = 168056704)